platform/core/src/ffi/java_interface.rs

```diff
@@ -16,15 +19,78 @@ pub extern "system" fn Java_coop_polypoly_core_JniApi_bootstrapCore(
     env: JNIEnv,
     _: JClass,
     language_code: JString,
+    callback: JObject,
 ) -> jbyteArray {
+    let bridge = BridgeToNative {
+        callback: env.new_global_ref(callback).unwrap(),
+        java_vm: env.get_java_vm().unwrap(),
+    };
     env.byte_array_from_slice(&serialize(
         read_jni_string(&env, language_code)
             .map(String::from)
-            .and_then(bootstrap),
+            .and_then(|language_code| bootstrap(language_code, Box::new(bridge))),
     ))
     .unwrap()
 }
 
+struct BridgeToNative {
+    // The callback passed from Android is a local reference: only valid during the method call.
+    // To store it, we need to put it in a global reference.
+    // See https://developer.android.com/training/articles/perf-jni#local-and-global-references
+    callback: GlobalRef,
+
+    // We need JNIEnv to call the callback.
+    // JNIEnv is valid only in the same thread, so we have to store the vm instead, and use it to get
+    // a JNIEnv for the current thread.
+    // See https://developer.android.com/training/articles/perf-jni#javavm-and-jnienvb
+    java_vm: JavaVM,
+}
+
+impl core::PlatformHookRequest for BridgeToNative {
+    fn perform_request(&self, request: NativeRequest) -> Result<NativeResponse, String> {
+        let result: Result<NativeResponse, String> = match self.java_vm.attach_current_thread() {
+            Ok(env) => {
+                let request_byte_array = env
+                    .byte_array_from_slice(&serialize(request))
+                    .map_err(|err| err.to_string())?;
+
+                let response_byte_array_as_jvalue = env
+                    .call_method(
+                        self.callback.as_obj(),
+                        "performRequest",
+                        "([B)[B",
+                        &[JValue::Object(JObject::from(request_byte_array))],
+                    )
+                    .map_err(|err| err.to_string())?;
+
+                let response_byte_array = response_byte_array_as_jvalue
+                    .l()
+                    .map_err(|err| err.to_string())?
+                    .into_inner();
+
+                let response_bytes: Vec<u8> = env
+                    .convert_byte_array(response_byte_array)
+                    .map_err(|err| err.to_string())?;
+
+                let deserialized: Result<NativeResponse, String> = deserialize(response_bytes)?;
+                deserialized
+            }
+            Err(e) => {
+                error!("Rust:java_interface => attach_current_thread:Err => Couldn't get env::");
+                Err(e.to_string())
+            }
+        };
+
+        if result.is_err() {
+            let err = result.err().unwrap();
+            error!("Rust:java_interface:perform_request => result:Err {}", err);
+            return Err(err);
+        } else {
+            return result;
+        }
+    }
```

Comment:
I this needed, why not return the result above? The error is handled at line 78.

Reply:
Line 78 logs to LogCat a custom error message for `attach_current_thread:Err`. 

If I have errors in `attach_current_thread:Ok` they won't appear in LogCat unless I catch them before I return the result. This will also catch the error returned by the system for `attach_current_thread:Err`.